### PR TITLE
[YUNIKORN-2577] Remove named returns from IsPodFitNodeViaPreemption

### PR DESCRIPTION
**Pull request by ryankert01**
### What is this PR for?
IsPodFitNodeViaPreemption has defined named returns but does not use them. They should be removed as the way they are used can cause issues that are hard to debug.

As part of this change we need to further cleanup:  
- The variable ok also gets shadowed multiple times, not just from the named return declaration.
- The if construct around GetPodNoLock() is not needed as it returns a nil for the pod if it returns false. Just adding the result for the pod always has the same effect.

**I did following changes:**
1. remove named returns
2. to avoid shadowing, I try not to use `ok`, instead, those function can represent `ok` by its first return value, whether if it's `index` or `pod`, `-1` and `nil` means `false` respectively.

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
N/A

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2577

### How should this be tested?
```
make test
```

### Screenshots (if appropriate)
N/A

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/cache/context.go [hunk 1]
```diff
@@ -681,17 +681,13 @@ func (ctx *Context) IsPodFitNodeViaPreemption(name, node string, allocations []s
 			// look up each victim in the scheduler cache
 			victims := make([]*v1.Pod, 0)
```
Comment (chia7712) [inline pkg/cache/context.go:682]: Could we set the size?

File: pkg/cache/context.go [hunk 2]
```diff
@@ -681,17 +681,13 @@ func (ctx *Context) IsPodFitNodeViaPreemption(name, node string, allocations []s
...
-					victims = append(victims, nil)
-				}
+				victim, _ := ctx.schedulerCache.GetPodNoLock(uid)
+				victims = append(victims, victim)
 			}
 
 			// check predicates for a match
-			if index, ok := ctx.predManager.PreemptionPredicates(pod, targetNode, victims, startIndex); ok {
-				return index, ok
+			if index, _ := ctx.predManager.PreemptionPredicates(pod, targetNode, victims, startIndex); index != -1 {
```
Comment (chia7712) [inline pkg/cache/context.go:689]: It seems `-1` is equal to "failure". Maybe we don't return extra `boolean` value.

**Review by chia7712 — COMMENTED** [review]
@ryankert01 thanks for this refactor. two minor comments left.

File: pkg/cache/context.go (hunk 2 above)
Comment (ryankert01) [reply pkg/cache/context.go:689]: It looks like we can refactor PreemptionPredicates(). I will file a jira for it.

File: pkg/cache/context.go (hunk 1 above)
Comment (ryankert01) [reply pkg/cache/context.go:682]: Yes, we can initialized the size first to further enhance the performance.

Comment (ryankert01) [reply pkg/cache/context.go]: Done.

File: pkg/cache/context.go (hunk 2 above)
Comment (wilfred-s) [reply pkg/cache/context.go:689]: The predicate manager also uses a mixture of named returns and just types. We should not use named returns there either. Filed a new jira for that one and a PR raised. The change to remove the boolean should be another jira.

**Review by wilfred-s — APPROVED** [review]
LGTM

File: pkg/cache/context.go [hunk 3]
```diff
@@ -663,35 +663,31 @@ func (ctx *Context) IsPodFitNode(name, node string, allocate bool) error {
 	return err
 }
 
-func (ctx *Context) IsPodFitNodeViaPreemption(name, node string, allocations []string, startIndex int) (index int, ok bool) {
+func (ctx *Context) IsPodFitNodeViaPreemption(name, node string, allocations []string, startIndex int) (int, bool) {
 	// assume minimal pods need killing if running in testing mode
 	if ctx.apiProvider.IsTestingMode() {
-		return startIndex, ok
+		return startIndex, false
```
Comment (chia7712) [inline pkg/cache/context.go:669]: This will get removed by #832, so we can revert this change to avoid conflicts later

Comment (chia7712) [reply pkg/cache/context.go:669]: I guess @wilfred-s is going to merge it, so please ignore my comment :)

**Pull request closed without merging**